org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java

```diff
@@ -1721,6 +1721,23 @@ public final class JavaCore extends Plugin {
 	 */
 	public static final String COMPILER_PB_UNLIKELY_EQUALS_ARGUMENT_TYPE = PLUGIN_ID + ".compiler.problem.unlikelyEqualsArgumentType"; //$NON-NLS-1$
 
+	/**
+	 * Compiler option ID: Reporting an equal comparison with dubious reference types, i.e. wrapper types (Integer, Long) for primitives (int, resp. long) and Strings.
+	 * <p>
+	 * When enabled, the compiler will issue an error or a warning if a comparison
+	 * is involving wrapper type (or String) operands (e.g <code>'1234L == 5678L'</code>).</p>
+	 * <p>
+	 * This warning can be suppressed using {@code @SuppressWarnings("reference-comparison")}.
+	 * <dl>
+	 * <dt>Option id:</dt><dd><code>"org.eclipse.jdt.core.compiler.problem.dubiousReferenceComparison"</code></dd>
+	 * <dt>Possible values:</dt><dd><code>{ "error", "warning", "info", "ignore" }</code></dd>
+	 * <dt>Default:</dt><dd><code>"info"</code></dd>
+	 * </dl>
+	 * @since 3.38
```

Comment:
When all code changes are done, we'll have to bump up bundles to 3.39.0 (currently 3.38.100) and mention that version in `@since` tags.
I can take care of that in the end.